amundsen_application/static/js/components/common/Badges/index.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import Flag, { FlagProps } from 'components/common/Flag';
+import { ResourceType } from 'interfaces';
+import { updateSearchState } from 'ducks/search/reducer';
+import { UpdateSearchStateRequest } from 'ducks/search/types';
+import { logClick } from 'ducks/utilMethods';
+
+import './styles.scss';
+
+export interface DispatchFromProps {
+  searchBadge: (badgeText: string) => UpdateSearchStateRequest;
+}
+
+export type ClickableBadgeProps = FlagProps & DispatchFromProps;
```

Comment:
Let's add a unit test for this file before wrapping up. When writing a unit test, I first like to think about what could be important to verify for the new component and/or updated behavior. For this component the following three cases come to mind:
1. Verify the component renders the `Flag` as expected, I think you've already seen examples of this in the `BadgeList` test.
2. We want to simulate a click event and verify the correct thing happens. [Example](https://github.com/lyft/amundsenfrontendlibrary/blob/ccfd2d6b82957fef347e956b243e4048c191fc0d/amundsen_application/static/js/components/common/Bookmark/BookmarkIcon/index.spec.tsx#L34-L44).
3. Verify the `mapDispatchToProps` functionality (as that is what makes the feature work!). We admittedly don't have good examples of this in the repo, but let me push an example with `TagInfo` for reference.